api/api/spec/spec.yaml

```diff
@@ -2570,291 +2661,347 @@ components:
+        format: alphanumeric
     address:
       in: query
       name: address
       description: Filters by IP address.
       schema:
         type: string
+        format: alphanumeric
     broadcast:
       in: query
       name: broadcast
       description: Filters by broadcast direction.
       schema:
         type: string
+        format: alphanumeric
     netmask:
       in: query
       name: netmask
       description: Filters by netmask.
       schema:
         type: string
+        format: alphanumeric
     adapter:
       in: query
       name: adapter
       description: Filters by adapter.
       schema:
         type: string
+        format: alphanumeric
     state:
       in: query
       name: state
       description: Filters by state.
       schema:
         type: string
+        format: alphanumeric
     mtu:
       in: query
       name: mtu
       description: Filters by mtu.
       schema:
         type: integer
+        format: int32
         minimum: 0
     tx_packets:
       in: query
       name: tx_packets
       description: Filters by tx_packets.
       schema:
         type: integer
+        format: int32
         minimum: 0
     rx_packets:
       in: query
       name: rx_packets
       description: Filters by rx_packets.
       schema:
         type: integer
+        format: int32
         minimum: 0
     tx_bytes:
       in: query
       name: tx_bytes
       description: Filters by tx_bytes.
       schema:
         type: integer
+        format: int32
         minimum: 0
     rx_bytes:
       in: query
       name: rx_bytes
       description: Filters by rx_bytes.
       schema:
         type: integer
+        format: int32
         minimum: 0
     tx_errors:
       in: query
       name: tx_errors
       description: Filters by tx_errors.
       schema:
         type: integer
+        format: int32
         minimum: 0
     rx_errors:
       in: query
       name: rx_errors
       description: Filters by rx_errors.
       schema:
         type: integer
+        format: int32
         minimum: 0
     tx_dropped:
       in: query
       name: tx_dropped
       description: Filters by tx_dropped.
       schema:
         type: integer
+        format: int32
         minimum: 0
     rx_dropped:
       in: query
       name: rx_dropped
       description: Filters by rx_dropped.
       schema:
         type: integer
+        format: int32
         minimum: 0
     iface:
       in: query
       name: iface
       description: Filters by network interface.
       schema:
         type: string
+        format: alphanumeric
     gateway:
       in: query
       name: gateway
       description: Filters by network gateway.
       schema:
         type: string
+        format: alphanumeric
     dhcp:
       in: query
       name: dhcp
       description: Filters by network dhcp (enabled or disabled).
       schema:
-        type: integer
+        type: string
+        enum:
+          - enabled
+          - disabled
```

Comment:
The `dhcp` field can have more values:
https://github.com/wazuh/wazuh/blob/8a34d85662154d324cd4c01455e205dc34367f04/src/wazuh_db/schema_agents.sql#L67

Reply:
I have refactored the dhcp schema definition. Now we have a unique definition reused both in parameter and SyscollectorProtocol schema.